modules/afsocket/afsocket-dest.c

```diff
@@ -552,8 +553,19 @@ afsocket_dd_setup_writer(AFSocketDestDriver *self)
       log_pipe_unref((LogPipe *) self->writer);
       return FALSE;
     }
-
   log_pipe_append(&self->super.super.super, (LogPipe *) self->writer);
+
+  if (kept_alive_connection)
+    {
+      LogProtoClient *proto = log_writer_steal_proto(self->writer);
+
+      if (proto)
+        {
+          self->fd = log_proto_client_get_fd(proto);
+          self->connection_initialized = TRUE;
+          log_writer_reopen(self->writer, proto);
+        }
```

Comment:
I am not sure this is easier to follow, as this puts an extra burden on the LogWriter's owner. 

We have similar reload logic in afprog for instance.